octavia_f5/restclient/as3objects/cipher.py

```diff
@@ -0,0 +1,91 @@
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from octavia.common import validate
+from octavia_f5.common import constants
+from octavia_f5.restclient.as3classes import Cipher_Rule, Cipher_Group
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+def get_cipher_rule_name(object_id, object_type):
+    """Returns AS3 object name for Cipher Rule related to a listener or a pool
+
+    :param object_id: octavia listener or pool id
+    :return: AS3 object name
+    """
+    return "{}{}_{}".format(constants.PREFIX_CIPHER_RULE, object_type.lower(), object_id)
+
+
+def get_cipher_group_name(object_id, object_type):
+    """Returns AS3 object name for Cipher Group related to a listener or a pool
+
+    :param object_id: octavia listener or pool id
+    :return: AS3 object name
+    """
+    return "{}{}_{}".format(constants.PREFIX_CIPHER_GROUP, object_type.lower(), object_id)
+
+
+def filter_cipher_suites(cipher_suites, object_name, object_id, http2=False):
+    """Filter out cipher suites according to blocklist and allowlist.
+
+    This is necessary, because there can be invalid cipher suites if e.g. a
+    previously allowed cipher suite was added to the blocklist recently and
+    listeners/pools using the cipher suite already existed.
+
+    :param cipher_suites: String containing colon-separated list of cipher suites.
+    :param object_name: A printable representation of the object to be logged, e.g. "Listener" or "Pool".
+    :param object_id: ID of the object the cipher suites belong to. This is used for logging, so it should be a string.
+    :param http2: HTTP2 protocol using
+    :return String containing colon-separated list of non-blocked/allowed cipher suites.
+    """
+
+    blocked_cipher_suites = validate.check_cipher_prohibit_list(cipher_suites)
+    disallowed_cipher_suites = validate.check_cipher_allow_list(cipher_suites)
+    rejected_cipher_suites = list(set(blocked_cipher_suites + disallowed_cipher_suites))
+
+    cipher_suites_list = cipher_suites.split(':')
+    if rejected_cipher_suites:
+        LOG.error("{} object with ID {} has invalid cipher suites which won't be provisioned: {}"
+                  .format(object_name, object_id, ', '.join(rejected_cipher_suites)))
+        for c in rejected_cipher_suites:
+            cipher_suites_list.remove(c)
+
+    # Add required ciphers if HTTP2 is using
+    for cipher in constants.CIPHERS_HTTP2:
+        if cipher not in cipher_suites_list:
+            LOG.warning("mandatory cipher {} was added for {} {} because HTTP2 is using"
+                        .format(cipher, object_name, object_id))
+            cipher_suites_list.append(cipher)
+
+    return cipher_suites_list
+
+
+def get_cipher_rule(ciphers, parent_obj, parent_id, http2=False):
```

Comment:
Please rename this method to something more correct, e. g. `get_cipher_rule_and_group`.